swifter/swifter.py

```diff
@@ -179,7 +179,6 @@ def resample(
         label=None,
         convention="start",
         kind=None,
-        loffset=None,
```

Comment:
If we are removing this due to the change in pandas API, maybe it's worth bumping the requirements for `pandas` to >1.1?

Reply:
pandas 1.1 did not change the acceptance of the optional parameter. It only introduced the deprecation at that point in time. Calling `resample()` will still work with older pandas versions, just not with using the `loffset` parameter.

Reply:
but does that mean that in pandas >1.1 that `loffset` can be submitted, but that it doesn't do anything?

Reply:
Looking at [this](https://pandas.pydata.org/pandas-docs/version/1.1/reference/api/pandas.DataFrame.resample.html), `loffset` is no longer an argument for `resample`, so I think it's going to raise a `TypeError` for the unexpected argument.

Reply:
It can be submitted to **pandas** until **<2.0** and has an effect, but will raise an error with **pandas >= 2.0**. *1.1* is not the defining epoch here.

It will also raise an error with **swifter** >= 1.4.0 as of this PR

Reply:
Here is my proposed solution for backwards compatibility https://github.com/jmcarpenter2/swifter/pull/220/commits/debf0ae9b44f4e89c673f530b8ed01c162e8c79c

Reply:
looks like 
![Screenshot 2023-07-24 at 10 02 09 AM](https://github.com/jmcarpenter2/swifter/assets/29585347/920393e7-2b85-4638-9353-620b38eec4a9)

Reply:
A simpler solution might be to pass the option through only if it's provided:
```python
        if loffset is not None:
                kwds.update({"loffset": loffset})
```

This results in:
- the signature here is unchanged (not perfect, but likely good enough)
- minimal additional code
- no need to understand pandas' versions (avoiding the issues raised in https://github.com/jmcarpenter2/swifter/pull/220#discussion_r1272634431)
- users get either the functionality, a deprecation warning or an error from pandas (depending on their version) (bear in mind that users will already have been getting deprecation warnings from pandas if they've been using a recent version)

You could optionally include a deprecation warning of your own if you wanted to, in case you wanted to remove the parameter yourself in future.

Reply:
Nice, i like it